src/handlers/swap.ts

```diff
@@ -114,18 +114,23 @@ export const getDefaultGasLimitForTrade = (
       tradeDetails?.sellTokenAddress?.toLowerCase() as keyof PermitSupportedTokenList
     ];
 
-  let defaultGasLimit = tradeDetails?.defaultGasLimit;
+  let socketGasLimit = tradeDetails?.defaultGasLimit;
+  const rainbowDefaultGasLimit =
+    ethereumUtils.getBasicSwapGasLimit(Number(chainId)) * EXTRA_GAS_PADDING;
 
   if (allowsPermit) {
-    defaultGasLimit = Math.max(
-      Number(defaultGasLimit),
+    socketGasLimit = Math.max(
+      Number(socketGasLimit),
       Number(ethUnits.basic_swap_permit) * EXTRA_GAS_PADDING
     ).toString();
   }
-  return (
-    Number(defaultGasLimit || 0) ||
-    ethereumUtils.getBasicSwapGasLimit(Number(chainId)) * EXTRA_GAS_PADDING
-  );
+
+  const defaultGasLimit = Number(socketGasLimit);
+  if (defaultGasLimit < rainbowDefaultGasLimit) {
```

Comment:
you could do `lessThan` instead of transforming wei values to number which is not ideal